crates/nassun/src/fetch/git.rs

```diff
@@ -219,3 +219,142 @@ impl PackageFetcher for GitFetcher {
+
+    use super::GitFetcher;
+
+    fn setup_git_dir() -> miette::Result<tempfile::TempDir> {
+        let git_dir = tempdir().unwrap();
+
+        process::Command::new("git")
+            .args(["init", "--initial-branch", "main"])
+            .current_dir(&git_dir)
+            .status()
+            .expect("Could not init git test directory");
+
+        let mut package_file = File::create(&git_dir.path().join("package.json")).unwrap();
+        package_file
+            .write_all(
+                r#"{
+            "name": "oro-test",
+            "version": "1.0.0"
+        }"#
+                .as_bytes(),
+            )
+            .unwrap();
+        drop(package_file);
+
+        // commit the first version
+        process::Command::new("git")
+            .args(["add", "package.json"])
+            .current_dir(&git_dir)
+            .status()
+            .expect("Could not add package.json to git repo");
+        process::Command::new("git")
+            .args(["commit", "-m", "First version", "--no-gpg-sign"])
+            .current_dir(&git_dir)
+            .status()
+            .expect("Could not commit first version");
+        process::Command::new("git")
+            .args(["tag", "--no-sign", "1.0.0"])
+            .current_dir(&git_dir)
+            .status()
+            .expect("Could not tag first version");
+
+        let mut package_file = File::create(&git_dir.path().join("package.json")).unwrap();
+        package_file
+            .write_all(
+                r#"{
+            "name": "oro-test",
+            "version": "1.2.0"
+        }"#
+                .as_bytes(),
+            )
+            .unwrap();
+        drop(package_file);
+
+        // commit the second version
+        process::Command::new("git")
+            .args(["commit", "-a", "-m", "Second version", "--no-gpg-sign"])
+            .current_dir(&git_dir)
+            .status()
+            .expect("Could not commit second version");
+        process::Command::new("git")
+            .args(["tag", "--no-sign", "1.2.0"])
+            .current_dir(&git_dir)
+            .status()
+            .expect("Could not tag first version");
+
+        let mut package_file = File::create(&git_dir.path().join("package.json")).unwrap();
+        package_file
+            .write_all(
+                r#"{
+            "name": "oro-test",
+            "version": "1.5.0"
+        }"#
+                .as_bytes(),
+            )
+            .unwrap();
+        drop(package_file);
+
+        // commit the third version
+        process::Command::new("git")
+            .args(["commit", "-a", "-m", "Second version", "--no-gpg-sign"])
+            .current_dir(&git_dir)
+            .status()
+            .expect("Could not commit second version");
+
+        return Ok(git_dir);
+    }
+
+    #[async_std::test]
+    async fn read_name() -> miette::Result<()> {
+        let git_dir = setup_git_dir()?;
+        // assert_eq!(name, "oro-test");
```

Comment:
wip?

Reply:
Fixed. Simply forgot to implement it while focusing on the packument test